Client/Frontend/Browser/Playlist/Cache/PlaylistCacheLoader.swift

```diff
@@ -0,0 +1,780 @@
+            return nil
+        }
+        
+        var alteredSource = source
+        let token = UserScriptManager.securityToken.uuidString.replacingOccurrences(of: "-", with: "", options: .literal)
+        
+        let replacements = [
+            "$<PlaylistDetector>": "PlaylistDetector_\(token)",
+            "$<security_token>": "\(token)",
+            "$<sendMessage>": "playlistDetector_sendMessage_\(token)",
+            "$<handler>": "playlistCacheLoader_\(UserScriptManager.messageHandlerTokenString)",
+            "$<notify>": "notify_\(token)",
+            "$<onLongPressActivated>": "onLongPressActivated_\(token)",
+            "$<setupLongPress>": "setupLongPress_\(token)",
+            "$<setupDetector>": "setupDetector_\(token)",
+            "$<notifyNodeSource>": "notifyNodeSource_\(token)",
+            "$<notifyNode>": "notifyNode_\(token)",
+            "$<observeNode>": "observeNode_\(token)",
+            "$<observeDocument>": "observeDocument_\(token)",
+            "$<observeDynamicElements>": "observeDynamicElements_\(token)",
+            "$<getAllVideoElements>": "getAllVideoElements_\(token)",
+            "$<getAllAudioElements>": "getAllAudioElements_\(token)",
+            "$<onReady>": "onReady_\(token)",
+            "$<observePage>": "observePage_\(token)",
+        ]
+        
+        replacements.forEach({
+            alteredSource = alteredSource.replacingOccurrences(of: $0.key, with: $0.value, options: .literal)
+        })
+        return WKUserScript(source: alteredSource, injectionTime: .atDocumentStart, forMainFrameOnly: false)
+    }()
+    
+    private var handler: (PlaylistInfo?) -> Void = { _ in }
+    private var handlerDidExecute = false
+    
+    init(handler: @escaping (PlaylistInfo?) -> Void) {
+        super.init(frame: .zero)
+
+        guard let webView = tab.webView else {
+            handlerDidExecute = true
+            handler(nil)
+            return
+        }
+        
+        self.handler = { [weak self] in
+            guard let self = self else { return }
+            self.handlerDidExecute = true
+            handler($0)
+        }
+        
+        self.addSubview(webView)
+        webView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+        
+        if let browserController = (UIApplication.shared.delegate as? AppDelegate)?.browserViewController {
+            let KVOs: [KVOConstants] = [
+                .estimatedProgress, .loading, .canGoBack,
+                .canGoForward, .URL, .title,
+                .hasOnlySecureContent, .serverTrust
+            ]
+            
+            browserController.tab(tab, didCreateWebView: webView)
+            KVOs.forEach { webView.removeObserver(browserController, forKeyPath: $0.rawValue) }
+            webView.scrollView.removeObserver(browserController.scrollController, forKeyPath: KVOConstants.contentSize.rawValue)
+        }
+        
+        webView.uiDelegate = nil
```

Comment:
Is this here because creating a `Tab` adds this stuff by default in BVC? If so, maybe mention it in a comment